pages/api/transactions/index.js

```diff
@@ -1,39 +1,62 @@
 import dbConnect from "@/db/connect";
 import Transaction from "@/db/models/Transaction";
+import { getServerSession } from "next-auth/next";
+import { authOptions } from "../auth/[...nextauth]";
+import { getToken } from "next-auth/jwt";
 
 export default async function handler(request, response) {
+  const session = await getServerSession(request, response, authOptions);
+  const token = await getToken({ req: request });
+  const userId = token?.sub;
```

Comment:
Can be deleted -> Or better replaced to: `userEmail=token?email;` and then be used in the code.